### PR TITLE
[CodeWhisperer] Not sending telemetry when users are not editing the file

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
Since we add documentListeners when the editor instance is created, when users are switching to another, the %code tracker is still sending %code telemetry, which will be 0 and it increase a lot of noise for reading these metric.

This PR aims to block these scenario from sending telemetry.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt [hunk 1]
```diff
@@ -111,14 +111,17 @@ abstract class CodeWhispererCodeCoverageTracker(
...
-            project = null,
-            acceptedTokensSize,
-            language,
-            percentage,
-            startTime.toString(),
-            totalTokensSize
-        )
+        // percentage == 0 means totalTokens == 0 and users are not editing the document, thus we shouldn't emit telemetry for this
+        percentage?.let { percentage ->
+            CodewhispererTelemetry.codePercentage(
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:116]: should this be a passive metric?

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:116]: ack, PR https://github.com/aws/aws-toolkit-common/pull/345

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt [hunk 2]
```diff
@@ -111,14 +111,17 @@ abstract class CodeWhispererCodeCoverageTracker(
...
 
-        CodewhispererTelemetry.codePercentage(
-            project = null,
-            acceptedTokensSize,
-            language,
-            percentage,
-            startTime.toString(),
-            totalTokensSize
-        )
+        // percentage == 0 means totalTokens == 0 and users are not editing the document, thus we shouldn't emit telemetry for this
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:114]: comment should be "percentage == null ...."

**Review by andrewyuq — APPROVED** [review]

**Review by abhikuhikar — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt (hunk 2 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:114]: oops let me update it

**Review by rli — APPROVED** [review]

**Pull request merged**